app/controllers/foreman_tasks/api/tasks_controller.rb

```diff
@@ -141,6 +141,31 @@ def bulk_cancel
         }
       end
 
+      api :POST, '/tasks/bulk_stop', N_('Stop all stoppable tasks')
+      param :search, String, :desc => N_('Stop tasks matching search string')
+      param :task_ids, Array, :desc => N_('Stop specific tasks by ID')
+      def bulk_stop
+        if params[:search].nil? && params[:task_ids].nil?
+          raise BadRequest, _('Please provide at least one of search or task_ids parameters in the request')
+        end
+
+        filtered_scope = bulk_scope
+        to_stop = filtered_scope.where.not(state: :stopped)
+        total_length = filtered_scope.count
+        to_stop_length = to_stop.length
+        to_stop.update_all(state: :stopped)
```

Comment:
`update_all` returns the number of records it updated, so you can merge those two lines into one

Reply:
changed to arrays so using length